### ISSUE TITLE
please provide a `install_torch()` option that relies on local `file:/` URL

### ISSUE THREAD

**Issue by cregouby** [mlverse/torch#436]
Hello, 

### Current situation
`install_torch()` fails on air-gap networked server, as mentionned by @dirkschumacher in https://github.com/mlverse/torch/issues/267#issuecomment-716711696_.

### Possible improvement
In order to install `libtorch` and `liblantern` on air-gapped server, we can usually rely on zip files uploaded on the machine filesystem. 

Could we have an `install_torch()` option or an alternative command like `install_torch_from_file()` that relies on local filesystem zip files ?

### Workaround
As a current hack to this, it is possible (but not convenient) to manually change the files URL to local files, as `utils::download.file()` that is used behind the scene supports `files://` prefixed url: 
1 - copy the repository `R/install.R` file in a local R session
2- edit the content of `install_config` list on line 4 with file urls turned into  `file://...` entries : 
In my very personnal case it turn out to be
```(r)
install_config <- list(
  "1.7.1" = list(
    "cpu" = list(
      "darwin" = list(
        "libtorch" = list(
          url = "file:///Users/to9444/Downloads/libtorch-macos-1.7.1.zip",
          path = "libtorch/lib",
          filter = ".dylib",
          md5hash = "d1d0561fa91326afcb7f0abe0241e704"
        ),
        "liblantern" = "file:///Users/to9444/Downloads/macOS.zip"
      )
    )
  )
)
```

3- perform install with current `install_torch()` command : 
```
> install_torch(reinstall = TRUE)
trying URL 'file:/Users/to9444/Downloads/libtorch-macos-1.7.1.zip'
Content type 'unknown' length 141774525 bytes (135.2 MB)
==================================================
trying URL 'file:/Users/to9444/Downloads/macOS.zip'
Content type 'unknown' length 1546872 bytes (1.5 MB)
==================================================

```

**Comment by dfalbel** [comment]
Hi @cregouby 

Thanks very much for the detailed report. I'll make sure that we support this in the next torch release.

**Comment by dfalbel** [comment]
Fixed in #439
